### PR TITLE
Add Machine Modal

### LINKED ISSUE #26: Allow users to add machines
As a technician, I want to add new machines to the application so that I can track their sensor data

**Acceptance Criteria**
 - If a user is logged in they can add a new machine for the application to track
 - When this machine is being added, the user can associate any number of sensors related to the machine
 - Once this machine is added, it can be viewed like the others from the overview page

**Is your feature request related to a problem? Please describe.**
Without this no machines can be added to the application without technical support from a developer. This would mean to keep the app functional in a meaningful and helpful way, constant maintenance would be required.

**Describe the solution you'd like**
There should be a button in the relevant place (e.g. on the machine overview page or profile page) which allows users to add sensors to that machine which can then be tracked

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### LINKED ISSUE #53: Bug: Back Button Disappearing
**Describe the bug**
- There's an issue with the back button missing whilst inside the sensors page inside a certain machine. (on the top left)
- It seems to disappear only when you access the sensors page from the back button on the individual sensor page.

**Acceptance Criteria**
 - Back button exists on every page to go to the previous page.

**To Reproduce**
- Click on one of the machines (can be dummyMachineOne).
- Click on one of the sensors inside the machine, to go to the individual sensor page
- Click on the back button inside the individual sensor page
- Then, you'll see that the back button is missing

**Expected behavior**
- There should be another back button to go back to the page before that.

**Screenshots**
Screen Shot 2020-09-28 at 1 00 29 AM

**Desktop (please complete the following information):**
 - OS: macOS 10.15.6
 - Browser: Safari, Firefox
 - Version: 14.0

**Additional context**
Taken from comment on PR #51

### PR DESCRIPTION
**Pull request by Deagler**
# **Review this PR After #47 **

## GitHub Issue Solved:

It's meant to close #26 but I think the AC on #26 overlaps with the AC for #21, #10, and #12  

Also closes #53 

## Current behaviour

- users cant add machines

## Changed behaviour

- users can add machines via a modal on the main page.

## PR checklist

Remember to check the following

 - [x] Tag specific people to review your PR
 - [x] Update the ReadMe with any new run instructions
 - [ ] Closes the associated issue (if applicable)

## Notes

image

### CODE REVIEW

**Review by FrankWangma — APPROVED** [review]
Adding machine works.
May require some integration with the other PRs at a later time

**Review by FrankWangma — APPROVED** [review]
Integration changes have been added. LGTM

File: app/src/components/HealthContainer.tsx [hunk 1]
```diff
@@ -10,8 +10,8 @@ interface ContainerProps {
 const HealthContainer: React.FC<ContainerProps> = ({ name, value, health }) => {
     const bg: string = statusColour(health);
     return (
-        <div className="flex justify-center text-black">
-            <div className={`rounded-lg shadow-xl font-bold text-lg responsive-width m-4 bg-${bg}-550`}>
+        <div className="flex w-full text-black">
+            <div className={`rounded-lg shadow-xl font-bold text-lg w-full flex-grow bg-${bg}-550`}>
```
Comment (merw957) [inline app/src/components/HealthContainer.tsx:14]: This change messes up the formatting for the sensors and individual sensor pages. Is there a reason this was changed?

File: app/src/components/LineGraph.tsx [hunk 2]
```diff
@@ -63,11 +63,11 @@ const LineGraph: React.FC<LineGraphProps> = ({ title, data, redThreshold, yellow
     });
 
     return (
-        <div className="flex justify-center text-black">
-            <div className="bg-white rounded-lg items-center pt-4 pb-4 responsive-width m-6">
+        <div className="flex text-black">
+            <div className="bg-white rounded-lg items-center pt-4 pb-4 max-w-lg flex-grow">
```
Comment (merw957) [inline app/src/components/LineGraph.tsx:67]: Same here

**Review by merw957 — CHANGES_REQUESTED** [review]
Just left a comment about the changes to the HealthContainer and LineGraph files. Looks good overall though 👍

**Review by merw957 — APPROVED** [review]
Formatting issues are fixed + fixes back button bug. LGTM 👍

**Pull request merged**